test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+        t2 = torch.eye(5) * 2
+        self.assertEqual(t1.tensor(), t2)
+        self.assertEqual(torch.mean(t1), torch.mean(t2))
+
+class TestOverrideSubTensor(TestCase):
+
+    def test_mm(self):
+        t = SubTensor([[1, 2], [1, 2]])
+        self.assertEqual(torch.mm(t, t), 0)
+
+
+class TestOverrideSubDiagonalTensor(TestCase):
+
+    def test_mean(self):
+        t1 = SubDiagonalTensor(5, 2)
+        t2 = 10 * torch.eye(5) * 2
+        self.assertEqual(t1.tensor() * 10, t2)
+        self.assertEqual(torch.mean(t1), torch.mean(t2))
+
+    def test_mm(self):
+        t1 = DiagonalTensor(5, 2)
+        t2 = SubDiagonalTensor(5, 2)
+        t3 = torch.eye(5) * 2
+        self.assertEqual(torch.mm(t1, t2), 1)
+        self.assertEqual(torch.mm(t2, t1), 1)
+        self.assertEqual(torch.mm(t3, t1), 0)
+        self.assertEqual(torch.mm(t1, t3), 0)
+        self.assertEqual(torch.mm(t3, t2), 1)
+        self.assertEqual(torch.mm(t2, t3), 1)
+
+IGNORED_TORCH_FUNCTIONS = (
+    # unique_dim isn't actually in the torch namespace but is a low-level
+    # kernel so needs to be included here
+    'unique_dim',
+    'typename',
+    'is_tensor',
+    'is_storage',
+    'set_default_tensor_type',
+    'set_rng_state',
+    'get_rng_state',
+    'manual_seed',
+    'initial_seed',
+    'seed',
+    'save',
+    'load',
+    'set_printoptions',
+    'fork',
+    'get_default_dtype',
+    'get_num_interop_threads',
+    'get_num_threads',
+    'import_ir_module',
+    'import_ir_module_from_buffer',
+    'is_anomaly_enabled',
+    'is_grad_enabled',
+    'merge_type_from_type_comment',
+    'parse_ir',
+    'parse_schema',
+    'parse_type_comment',
+    'set_anomaly_enabled',
+    'set_flush_denormal',
+    'set_num_interop_threads',
+    'set_num_threads',
+    'wait',
+    'as_tensor',
+    'from_numpy',
+    'get_device',
+    'tensor',
+    'default_generator',
+    'has_cuda',
+    'has_cudnn',
+    'has_lapack',
+    'cpp',
+    'device',
+    'dtype',
+    'finfo',
+    'has_mkl',
+    'has_mkldnn',
+    'has_openmp',
+    'iinfo',
+    'memory_format',
+    'qscheme',
+    'set_grad_enabled',
+    'no_grad',
+    'enable_grad',
+    'layout',
+    'align_tensors',
+    'arange',
+    'as_strided',
+    'bartlett_window',
+    'blackman_window',
+    'can_cast',
+    'cudnn_affine_grid_generator',
+    'cudnn_batch_norm',
+    'cudnn_convolution',
+    'cudnn_convolution_transpose',
+    'cudnn_grid_sampler',
+    'cudnn_is_acceptable',
+    'empty',
+    'empty_strided',
+    'eye',
+    'from_file',
+    'full',
+    'hamming_window',
+    'hann_window',
+    'linspace',
+    'logspace',
+    'mkldnn_adaptive_avg_pool2d',
+    'mkldnn_convolution',
+    'mkldnn_convolution_backward_weights',
+    'mkldnn_max_pool2d',
+    'ones',
+    'promote_types',
+    'rand',
+    'randn',
+    'randint',
+    'randperm',
+    'range',
+    'zeros',
+)
+
+TENSOR_LIKE_TORCH_IMPLEMENTATIONS = (
+    (torch.abs, lambda input, out=None: -1),
```

Comment:
I see that a lot of these are `-1`. Any reason to not put the implementation as `lambda *a, **kw: -1`? And just run a loop over all possible functions?